lib/frr_pthread.c

```diff
@@ -210,11 +229,13 @@ void frr_pthread_stop_all()
 	pthread_mutex_unlock(&frr_pthread_hash_mtx);
 }
 
-unsigned int frr_pthread_get_id()
+uint32_t frr_pthread_get_id(void)
 {
+	_Atomic uint32_t nxid;
+	nxid = atomic_fetch_add_explicit(&next_id, 1, memory_order_seq_cst);
 	/* just a sanity check, this should never happen */
-	assert(next_id <= INT_MAX - 1);
-	return next_id++;
+	assert(nxid <= (INT_MAX - 1));
```

Comment:
nit: `INT_MAX` is 32767. Couldn't we use `UINT32_MAX` (4294967295) here? It might make a difference if a daemon spawns short-lived threads on a regular base.

Reply:
yep this is a mistake, thanks